### ISSUE TITLE
(Voicing) Optimizing the Leg Swing interaction and accumulation of electrons 

### ISSUE THREAD

**Issue by terracoda** [phetsims/john-travoltage#397]
More details coming, but for non-visual access we can **NOT** have blank spots (no sound and no description) on the Leg Swing interaction.

I think we can get away with adding in object responses (like aria-valuetext position descriptions) just for the "off rug" positions. When on the rug, learners get sonification and electron responses, but when off rug, they get nothing.

@Matthew-Moore240, do you agree?

**Comment by terracoda** [comment]
I think this is good path forward as we know this description works, and we don't have a ready sound to use instead.

**Comment by terracoda** [comment]
Ok, I think I figured out why the leg interaction seems to lag so much. It's because we changed how we announce the information during a drag. I think originally we announced new position at end of drag and only the name of the slider at the start of the drag. 

I need to review things, but first lets just add "foot off rug" position descriptions, and remove "Rubbing" from the accumulation of electrons response.

Once that is done, I can figure out when we are supposed to hear position info and when we hear electron accumulation.

**Comment by Matthew-Moore240** [comment]
Yeah, this looks good to me! Handing off to @jessegreenberg

**Comment by terracoda** [comment]
@jessegreenberg, for positions, please add to all positions that do currently create a rubbing sound. If you start from the back position, this is how I see it:
1. foot off rug, pointing backward
2. foot off rug
3. foot off rug
4. foot off rug
5. foot on rug - silent during interaction, but announced on focus 
6. foot on rug - silent during interaction, but announced on focus
7. foot on rug - silent during interaction, but announced on focus
8. foot on rug - silent during interaction, but announced on focus
9. foot on rug - silent during interaction, but announced on focus
10. foot on rug - silent during interaction, but announced on focus
11. foot off rug
12. foot off rug
13. foot off rug
14. foot off rug
15. foot off rug, pointing forward

**Example focus responses:**
1. Leg Swing, foot off rug, pointing backward
4. Leg Swing, foot off rug
7. Leg Swing, foot on rug 
15. Leg Swing, foot off rug, pointing forward

**Example non-focus interaction responses:**
1. foot off rug, pointing backward
4.foot off rug
7. a couple electrons on body (no position spoken for foot on rug)
15. foot off rug, pointing forward

**Comment by jessegreenberg** [comment]
Finally looking into this...

The reason we have blank spots is because we decided specifically not to alert leg positions while dragging it in #377.
> The leg position is only described on pickup/release. Otherwise, user only hears about charge pickup during drag.

I tried to just add the leg position alert, but it often be spoken over charge alert information when the drag ends with the foot off the rug.

**Comment by terracoda** [comment]
@jessegreenberg, thanks for linking in #377 . I should have done that originally. That was indeed the issue where we made several design changes, some of which had unintended side effects.

The issue is that learners have no voicing and no sonification when foot is off the rug.

Yes, it is possible to still possibly get a delayed electron response with foot off rug, but we are trying to address the place where successive steps in a row the foot is off the rug.

In https://github.com/phetsims/john-travoltage/issues/397#issuecomment-809340382, I am suggesting we add "foot off rug" to the off-positions only. I agree the on-positions naturally get get over-spoken anyways.

Is it possible to just add foot position (of which there are 4 unique ones) just to the off-positions and not add on-positions during active rubbing?

**Comment by jessegreenberg** [comment]
I went with the following logic which should preserve the behavior of the arm responses from https://github.com/phetsims/john-travoltage/issues/377#issuecomment-800576319, but allow leg responses to come through after charge responses.

```
if isLeg
  if shoeOnCarpet
    alert foot position politely (to allow charge alerts to come through)
else
  alert hand position assertively
```

**Comment by jessegreenberg** [comment]
This has been done in https://phet-dev.colorado.edu/html/john-travoltage/1.6.0-preferences.1/phet/john-travoltage_en_phet.html?supportsVoicing, @terracoda can you please review to make sure this is behaving as you would expect?

**Comment by terracoda** [comment]
The foot sounds great, so you have addressed this issue. Closing.
